Update WW3 for fix for MPI reproducibility

JessicaMeixner-NOAA opened the pull request:
# PR Checklist

- [x] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [x] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This PR updates WW3 for a fix to the MPI reproducibility issue (#427 https://github.com/NOAA-EMC/WW3/pull/518).  The issue was that if ice is updated throughout the run and you are using IC0 (which we are in UFS) points that become ice covered beyond a threshold are "deactivated" however, when using MPI, these points values were not being properly zero'd out as expected.  This has been fixed along with two other bug fixes (https://github.com/NOAA-EMC/WW3/pull/512 https://github.com/NOAA-EMC/WW3/pull/519) needed to run WW3 in debug mode (not completely finished or included in this PR, but these two required bug fixes are).  Answers will therefore change for wave tests which were effected by this bug (S2SW tests, HAFS tests with waves did not change answers).  The "cpld_bmark_mpi_p7" test was added to show that this bug was fixed as it is not apparent in the "cpld_control_wave_p7" test.  In addition to continue to test that WW3 is thread safe and passes some of the other tests, cpld_decomp_wave_p7, cpld_mpi_wave_p7, and cpld_2threads_wave_p7 were added.  

This PR also uses a new WW3 input directory which reduced the number of spectral points used in WW3 low resolution 1 deg grid  (#822) which in turn reduces the computational requirements for tests using the 1 deg grid.  Tests using this grid were profiled using ESMF on hera and resources were adjusted.  Additionally the ATMW tests were set-up to have machine dependent resources.   Any test using the 1 deg wave grid will have answer changes.   

A 5 deg grid and it's documentation is included in the new WW3 input that can be used for future low resolution tests. 
Lastly, the 1deg, 2deg and 5 deg tests have been updated to work for MOM6 coupling (issue #913), by making sure the namelist option: 
&OUTS USSP = 1, IUSSP = 3, STK_WN = 0.04, 0.110, 0.3305 /
was included in the appropriate ww3_grid.inp files. 

The new input that its required is currently staged on hera at: /scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20210930/WW3_input_data_20211113 (updated from original PR)

Co-authors: @aliabdolali @MatthewMasarik-NOAA

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
- fixes #427
- fixes #822
- fixes #913

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

This was tested in earlier versions on multiple platforms (dell, orion, gaea, hera.intel), with the most recent code a baseline was created and shown to match using hera.intel

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature: skip until oRTs can be updated to use S2SW
- [ ] CI: skip due to changes in this PR

## Dependencies

- waiting on https://github.com/NOAA-EMC/WW3/pull/521

DeniseWorthen commented:
The number of tests in this PR needs to be reduced. The current default cpld P7 test suite is for 3-components only because we could not turn waves on for the tested features (decomp, threading, restart). With the fixes for WW3, the existing cpld P7 test suite needs to be updated to include all 4 components. 

A separate 3-component test pair will need to be maintained only for restart reproducibility. The default_vars for the cpld_dflt should reflect the 4-component configuration, with only the 3-component c96mx100 configuration requiring a separate "non-wave" setting.

JessicaMeixner-NOAA commented:
Would an acceptable alternative be to turn off the tests that the new tests could replace instead of changing the defaults, etc?

DeniseWorthen commented:
The problem w/ that solution is that it implies the "default" is something other than a 4-component configuration.

JessicaMeixner-NOAA commented:
If "cpld_control_wave_p7" becomes the default with all the default settings I'm assuming we rename this test to then be "cpld_control_p7" but "cpld_control_p7" still needs to be run as part of "cpld_restart_p7" what does "cpld_control_p7" then become?  

Or am I just changing the values within the tests and replacing "cpld_2threads_wave_p7" with "cpld_2threads_p7" (same for decomp)?

DeniseWorthen commented:
I would suggest that, like the c192 and c384 test pairs, the non-wave, c96mx100 test that will be used for restart testing would be ``cpld_control_c96_p7`` and ``cpld_restart_c96_p7``. 

The default vars used for those tests would also be named in the same way as for the c192 and c384 cases: eg ``TASKS_cpl_c96`` 

All the tests currently in the ``cpld_control_p7`` set (decomp, etc) would keep the same name, but they would now have waves-on. 

The default vars should also be updated to make waves "on" to be the default. Those settings would then need to be explictly turned off in the test/restart pairs.

junwang-noaa commented:
cpld_control_wave_p7 will become cpld_control_p7 with 4 components, and the cpld_control_p7 becomes the cpld_control_p7_nowav, the restart test cpld_restart_p7_nowav (current cpld_restart_p7) is associated with it.

junwang-noaa commented on tests/default_vars.sh:
```diff
@@ -23,13 +23,35 @@ elif [[ $MACHINE_ID = wcoss_dell_p3 ]]; then
+
+  TASKS_cpl_thrd_wwav=120; TPN_cpl_thrd_wwav=14; INPES_cpl_thrd_wwav=3; JNPES_cpl_thrd_wwav=4
+  THRD_cpl_thrd_wwav=2; WPG_cpl_thrd_wwav=6;  MPB_cpl_thrd_wwav="0 71"; APB_cpl_thrd_wwav="0 77"
+  OPB_cpl_thrd_wwav="78 97"; IPB_cpl_thrd_wwav="98 107"; WPB_cpl_thrd_wwav="108 119"
```
OCN is still taking 20 tasks, does this mean OCN does not have threading turned on?

JessicaMeixner-NOAA replied:
@jiandewang can you comment on this question from Jun about MOM6 threading. 

@junwang-noaa there is about to be a fairly large update to this PR based on requests for changes, FYI.  Just doing the final testing now.

jiandewang replied:
@junwang-noaa threading in OCN won't gain speed, thus we are not using this

junwang-noaa replied:
@jiandewang Do you mean using threads in OCN does not speed up the OCN computation or it does not speed up the total run time for the test?

JessicaMeixner-NOAA commented:
This PR has now been updated to reduce the number of tests and have waves included in the coupled defaults.   @DeniseWorthen and @junwang-noaa I believe this is to your specifications.  Let me know if I misunderstood anything. 

We will also be updating the WW3 submodule pointer here soon to include an update for jet compilation flags.

DeniseWorthen commented:
Why are all the ``NPROC_ICE_ice`` settings required in ``default_vars``? They all look identical except for two settings on dell-p3.

JessicaMeixner-NOAA commented:
> Why are all the `NPROC_ICE_ice` settings required in `default_vars`? They all look identical except for two settings on dell-p3.

By explicitly setting this value it can 1. be different for different platforms and 2. can be straightforward for updating that number of pets in the future.   I think I had originally updated this value a little more than in this end product, but I think keeping it and that future flexibility could be nice.

DeniseWorthen commented:
If the desire is flexibility (different on different platforms), that argument could be made for any component. Ice actually seems like the least likely component to want to vary in this way, since it is generally thought to be the "cheapest". 

We have talked about setting a default tasks for each resolution and having the PE bounds calculated automatically. But setting nproc_ice still requires you to manually edit the PE bounds so I don't think I see the usefulness. It just means you have to change two things instead of one for each platform and test.

JessicaMeixner-NOAA commented:
Yes, the ice component is generally the cheapest, which is why it was also used at times to take a few processors from to make the full pet count use an entire node and not just a few more pets on an additional node.   If these extra NPROC_ICE variables are desired to be removed, it will take me another day or so to re-do the profiling on hera with the desired number of tasks for ice and then based on that re-do the counts for other components to either keep the same total number of tasks, etc.  Just let me know if that work is required.

DeniseWorthen commented:
I can see the point of using it to make up a full node. 

In ``cpld_bmark_p7`` ,  you are still setting NPROC_ICE to 48; there is no ``NPROC_ICE_cpl_bmrk`` in default_vars.

DeniseWorthen commented on tests/tests/cpld_control_c192_p7:
```diff
@@ -122,4 +122,11 @@ export FNVMXC="'C192.vegetation_greenness.tileX.nc'"
+
+export NEMS_CONFIGURE=nems.configure.cpld.IN
+
+
```
remove extra blank line

DeniseWorthen commented on tests/tests/cpld_control_c96_p7:
```diff
@@ -0,0 +1,89 @@
+#
+#  cpld_control C96 restart P7 test
```
remove 'restart'. This is the control test.

DeniseWorthen commented on tests/tests/cpld_debug_p7:
```diff
@@ -63,4 +63,28 @@ export FHMAX=6
+
+export NEMS_CONFIGURE=nems.configure.cpld.IN
+
+
```
another extra blank line

DeniseWorthen commented on tests/rt.conf:
```diff
@@ -3,23 +3,25 @@
+RUN     | cpld_mpi_p7                                                                                                             | - wcoss_cray                            |     |
+
+RUN     | cpld_bmark_p7                                                                                                           | - wcoss_cray                            | fv3 |
+RUN     | cpld_bmark_mpi_p7                                                                                                       | - wcoss_cray                            |     |
```
Why do we need an mpi test for both the default and bmark resolutions?

JessicaMeixner-NOAA replied:
The "cpld_bmark_mpi_p7" test was added to show that this bug was fixed as it is not apparent in the "cpld_control_wave_p7" test.  Without this test we could not debug the stated issue or catch the problem.  That being said, it's not typical to have this test.  It can be removed if deemed necessary for the final PR.

DeniseWorthen replied:
I read through the issue [317](https://github.com/NOAA-EMC/WW3/issues/317) and it isn't clear to me whether the explanation for why this didn't impact the low-resolution test but did impact the bmark configuration was found. I'm wondering if testing at low-resolution will be adequate?

JessicaMeixner-NOAA replied:
I believe we just got lucky that we do not see it on the C96 test case, essentially during the forecast time, we do not have a point that has enough ice growth during the forecast to become a "deactivated" sea-point and therefore trigger the issue. It was either a little luck, forecast length, IC date or function of resolution or all of the above.

DeniseWorthen replied:
So is the MPI test for both resolutions required, in your opinion?

JessicaMeixner-NOAA replied:
I assumed I'd be asked to delete this test, but wanted to include it in the initial PR at the minimum to show we accomplished our goal and on the off-chance others thought it should stay.  Testings non C96 low resolution for the operational requirements is not a part of the regression tests and I understand we have to draw the line of when/where to add the tests.  I also think that if the operational requirements test was expanded to support running the higher resolution cases, that could be sufficient.   I think the biggest argument for inclusion of this would be that different physics are used in the ocean model, but that's never been enough to add one in the past, so I don't see the wave model's situation would pass that threshold either.

DeniseWorthen replied:
OK, thanks. I have the same feeling about the longer c192 test; we initially created a long test for a specific issue that Moorthi had reported, but we've continued to carry it. It is hard to drop a test once it has been added. 

I think it might beIt useful in this case to commit the test just to mark a place where we knew something was working.

DeniseWorthen commented on tests/tests/control_c384gdas_wav:
```diff
@@ -54,33 +54,23 @@ export LIST_FILES="sfcf000.nc \
 export NPZP=128
 export NPX=385
 export NPY=385
 export DT_ATMOS=225
```
Does dt_atmos need to be 225 for this test? I know it is the current value but I wonder if it is actually required to be smaller than what we use for the coupled model.

JessicaMeixner-NOAA replied:
This is the same DT_ATMOS as in the atm-only case, which this test is based on: https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/tests/control_c384gdas#L64

DeniseWorthen commented on tests/tests/cpld_mpi_p7:
```diff
@@ -62,23 +62,25 @@ export_cpl
-export CPLWAV2ATM=.true.
-export eps_imesh=2.5e-1
+export NPROC_ICE=$NPROC_ICE_cpl_mpi
+
```
extra line

JessicaMeixner-NOAA commented:
> I can see the point of using it to make up a full node.
> 
> In `cpld_bmark_p7` , you are still setting NPROC_ICE to 48; there is no `NPROC_ICE_cpl_bmrk` in default_vars.

Updated!

junwang-noaa commented on tests/tests/cpld_2threads_p7:
```diff
@@ -51,11 +51,16 @@ export LIST_FILES="sfcf024.tile1.nc \
 
 export_fv3
 export_cpl
+export FHMAX=12
```
Is there any issue with 24 hr forecast length?

JessicaMeixner-NOAA replied:
The FHMAX=12 was in the old cpld_control_wave_p7 test and therefore carried over in the new cpld_control_p7 (with waves) and the associated tests.  This could certainly be changed if needed.

DeniseWorthen replied:
I thought you had perhaps changed FHMAX for timing reasons. Shorter tests are generally better, but 24h is our standard length. What do you think @junwang-noaa ?

JessicaMeixner-NOAA replied:
I think it was shortened in the cpld_control_wave_p7 test  a while ago for timing reasons; however, with the lighter WW3 grid, I don't think you'll have the timing issues for the reason it was changed to 12 (assuming my memory is correct, I didn't go back and look up the commit histories to confirm).

junwang-noaa replied:
Let's change it back to 24hrs, the 12hr forecast with the lighter ww3 grid now takes 2mins, so I don't think it would be an issue.

junwang-noaa commented on tests/default_vars.sh:
```diff
@@ -42,6 +61,12 @@ elif [[ $MACHINE_ID = wcoss_dell_p3 ]]; then
   OPB_cpl_bmrk="312 431"; IPB_cpl_bmrk="432 479"; WPB_cpl_bmrk="480 559"
+  NPROC_ICE_cpl_bmrk=48
+
+  TASKS_cpl_bmrk_mpi=600; TPN_cpl_bmrk_mpi=40; INPES_cpl_bmrk_mpi=6; JNPES_cpl_bmrk_mpi=8
```
The cpl_bmrk_mpi test only has wave tasks changed. I am wondering which test, cpld_bmark_p7 or cpld_bmark_mpi_p7, has better load balance?

JessicaMeixner-NOAA replied:
In the end, I was simplifying the changes to narrow them to just WW3 --- as that was the focus of the debug.  If this test is left in, I can change other pet tasks if desired. 

I did not do any load balancing on the cpl benchmark test as the wave grid did not change.  I assumed it was load-balanced as is and the other was just for testing the wave MPI issues.

DeniseWorthen commented on tests/default_vars.sh:
```diff
@@ -42,6 +61,12 @@ elif [[ $MACHINE_ID = wcoss_dell_p3 ]]; then
+  NPROC_ICE_cpl_bmrk_mpi=48
 
   TASKS_cpl_c192=288; TPN_cpl_c192=28; INPES_cpl_c192=4; JNPES_cpl_c192=8
   THRD_cpl_c192=1; WPG_cpl_c192=12;  MPB_cpl_c192="0 191"; APB_cpl_c192="0 203"
```
There are no settings for NPROC_ICE for any coupled c192 test or c384 test. If we're setting this in default_vars, we should be consistent.

Actually, I think that the issue of setting tasking in default_vars is out of control. It think we need to address it, but that is beyond the scope of this PR (Issue [320](https://github.com/ufs-community/ufs-weather-model/issues/320)).

JessicaMeixner-NOAA replied:
I added NPROC_ICE for c192/c384 for consistency.

JessicaMeixner-NOAA commented:
I believe I have addressed all the existing reviewers comments.  I'm recreating a baseline for the cpld tests that I changed to 24 hours and then will do another check to make sure I can replicate that baseline and do not have any other typos. I'll report back.

JessicaMeixner-NOAA commented:
I believe I have addressed all the existing reviewers comments.  I'm recreating a baseline for the cpld tests that I changed to  FHMAX=24 hours and then will do another check to make sure I can replicate that baseline and do not have any other typos. I'll report back.

DeniseWorthen commented:
Thanks Jessica for all the changes.  I've been over them again and I see no remaining issues. Hopefully we can start the baselines/commit tomorrow after your final testing.

junwang-noaa commented:
@JessicaMeixner-NOAA Thanks for addressing all the issues. The code changes look good to me. Please let us know how the hera testing is going, we will then sync the input data and start RT.

junwang-noaa commented:
@MinsukJi-NOAA The cpld_control_p7 test is updated to include wave in this PR, CI test may need to be updated too.

JessicaMeixner-NOAA commented:
The input data should be fine to sync...  Before running the full tests I'm trying to figure out one error, which I can't seem to  make sense of for cpld_control_p7: 

```
 Comparing RESTART/ufs.cpld.cpl.r.2021-03-23-21600.nc .........OK
 Comparing \ ............MISSING file
 Comparing 20210323.060000.out_grd.glo_1deg .........OK
```

The cpld_bmark_mpi_p7 test also just failed in my last round of testing but that was an srun hang error not actual model error.

junwang-noaa commented:
It seems to me there might be some extra space in the file list for
comparison in cpld_control_p7 test?

On Mon, Nov 15, 2021 at 7:39 PM Jessica Meixner ***@***.***>
wrote:

> The input data should be fine to sync... Before running the full tests I'm
> trying to figure out one error, which I can't seem to make sense of for
> cpld_control_p7:
>
>  Comparing RESTART/ufs.cpld.cpl.r.2021-03-23-21600.nc .........OK
>  Comparing \ ............MISSING file
>  Comparing 20210323.060000.out_grd.glo_1deg .........OK
>
> The cpld_bmark_mpi_p7 test also just failed in my last round of testing
> but that was an srun hang error not actual model error.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/911#issuecomment-969522414>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPBBT2IYYAISODWFVDUMGR5JANCNFSM5H43RGDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

junwang-noaa commented:
Input data directory WW3_input_data_20211113 has been synced to all the platforms except Cheyenne. @aliabdolali Do we still need the WW3_input_data_20211112, if not, would you please delete it on Orion? Thanks.

aliabdolali commented:
> Input data directory WW3_input_data_20211113 has been synced to all the platforms except Cheyenne. @aliabdolali Do we still need the WW3_input_data_20211112, if not, would you please delete it on Orion? Thanks.

Done,

DeniseWorthen commented:
The WW3_input_data_20211113 has been added to input-data-20210930 on Cheyenne. 

On Hera, what is input-data-directory-20211115? Was that directory and/or develop-202112XX created in association with this PR?

JessicaMeixner-NOAA commented:
I did not create the input-data-directory-20211115 or  develop-202112XX  directory.  I only created: 
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20210930/WW3_input_data_20211112 (Which is no longer needed, and Ali deleted the copy he made on orion)  and
/scratch1/NCEPDEV/nems/emc.nemspara/RT/NEMSfv3gfs/input-data-20210930/WW3_input_data_20211113

I do have a baseline on hera for intel here: /scratch1/NCEPDEV/stmp4/Jessica.Meixner/FV3_RT/Baseline20111115 if we want to copy that instead of re-creating. 

The regression tests passed last night after deleting an extra space in the cpld_control_p7 list of files to compare.

JessicaMeixner-NOAA commented:
I updated the baseline date to today's date.

junwang-noaa commented:
I created the input-data-20211015 on hera to remove the
diag_table/field_table in the input directory and use a central version of
CO2 data for all the RT tests

On Tue, Nov 16, 2021 at 6:56 AM Jessica Meixner ***@***.***>
wrote:

> I updated the baseline date to today's date.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/911#issuecomment-970199057>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TPFWPVPPQKVC5NVS63UMJBFDANCNFSM5H43RGDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: hera
Compiler: gnu
Job: BL
Repo location: /scratch1/NCEPDEV/nems/emc.nemspara/autort/pr/779412862/20211116144510/ufs-weather-model
Please manually delete: /scratch1/NCEPDEV/stmp2/emc.nemspara/FV3_RT/rt_10187
Test cpld_control_p7 031 failed failed 
Test cpld_control_p7 031 failed in run_test failed 
Please make changes and add the following label back:
hera-gnu-BL

JessicaMeixner-NOAA commented:
The failure for the gnu test is because I did not change to use the non-wave control test. That fix was pushed.

DeniseWorthen commented:
@JessicaMeixner-NOAA Will this PR also close Issue #427 ?

(EDIT) Sorry--I was confused by the two different titles on the issues vs PR. No answer required.

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: cheyenne
Compiler: intel
Job: BL
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/779412862/20211116094511/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_16213
Baseline creation and move successful
Repo location: /glade/scratch/dtcufsrt/autort/tests/auto/pr/779412862/20211116105416/ufs-weather-model
Please manually delete: /glade/scratch/dtcufsrt/FV3_RT/rt_59390
Test cpld_bmark_mpi_p7 006 failed failed 
Test cpld_bmark_mpi_p7 006 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-intel-BL

DeniseWorthen commented:
I can't spot any obvious reason why the mpi_p7 test failed. I will try running separately.

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/779412862/20211116163017/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_18590
Baseline creation and move successful
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/779412862/20211116181538/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_47060
Test cpld_bmark_mpi_p7 006 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

JessicaMeixner-NOAA commented:
@DeniseWorthen I think the Cheyenne and gaea test failed for the same reason gaea did: 
TPN_cpl_bmrk_mpi=40 not TPN_cpl_bmrk_mpi=36 for each.

JessicaMeixner-NOAA commented:
> @DeniseWorthen I think the Cheyenne and gaea test failed for the same reason gaea did:
> TPN_cpl_bmrk_mpi=40 not TPN_cpl_bmrk_mpi=36 for each.

Although I just saw you pushed the Cheyenne log.  Should I make this change or not?

DeniseWorthen commented:
@JessicaMeixner-NOAA Please make the change for gaea only while I think about what to do w/ Cheyenne.

JessicaMeixner-NOAA commented:
@DeniseWorthen I made the update for gaea, sorry about the issue on both machines for this test.

DeniseWorthen commented:
@JessicaMeixner-NOAA This sort of error is all too easy w/ the number of platforms and number of tasking options w/ the coupled model. 

Please go ahead and update the Cheyenne tasks. I will re-run that test and make an empty commit that tests passes a second time.

BrianCurtis-NOAA commented:
***Automated RT Failure Notification***
Machine: jet
Compiler: intel
Job: BL
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/779412862/20211116163015/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_168928
Baseline creation and move successful
Repo location: /lfs4/HFIP/h-nems/emc.nemspara/autort/pr/779412862/20211116182248/ufs-weather-model
Please manually delete: /lfs4/HFIP/h-nems/emc.nemspara/RT_RUNDIRS/emc.nemspara/FV3_RT/rt_224474
Test cpld_bmark_mpi_p7 005 failed in run_test failed 
Please make changes and add the following label back:
jet-intel-BL

JessicaMeixner-NOAA commented:
Are we waiting on the CI or should we go ahead and start the process of merging the WW3 submodule?

DeniseWorthen commented:
@MinsukJi-NOAA CI will need an update in the next PR, is that right?

MinsukJi-NOAA commented:
> @MinsukJi-NOAA CI will need an update in the next PR, is that right?

I suggest we skip CI for this PR; it will be fixed in the next PR.

junwang-noaa commented:
Thanks, Minsuk. Let's start the commit process.

On Wed, Nov 17, 2021 at 9:10 AM Minsuk Ji ***@***.***> wrote:

> @MinsukJi-NOAA <https://github.com/MinsukJi-NOAA> CI will need an update
> in the next PR, is that right?
>
> I suggest we skip CI for this PR; it will be fixed in the next PR.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/911#issuecomment-971617894>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AI7D6TLPPUCARMZHXPDV6STUMOZVHANCNFSM5H43RGDA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>

aliabdolali commented:
thank you all for doing this.
